### PR TITLE
Documentation workflow based on Kindly & Clay

### PR DESCRIPTION
**Pull request by daslu**
**edit: corrected parhs**

## New experimental dev workflow

This PR is part of our work on a new dev workflow for Tablecloth:
- namespace-as-a-notebook documentation using [Kindly](https://scicloj.github.io/kindly) and [Clay](https://scicloj.github.io/clay) -- this PR
- testing the documentation using [note-to-test](https://github.com/scicloj/note-to-test) -- coming soon

### Status
 **Not to be merged yet.** 
 There are still some rendering issues (e.g. in the layout of R code output).

### Relevant files
- `notebooks/draft.clj` - the tutorial as a Kindly notebook (developed with Clay)
  - `dev/conversion.clj` - the script used to generate the notebook from the original `Rmarkdown` tutorial (up to a few additional manual edits)
- `docs/draft.html` - the tutorial rendered using Clay and [Quarto](https://quarto.org/) -- temporarily served at [https://daslu.github.io/tablecloth/draft](https://daslu.github.io/tablecloth/draft)

### Actions
- to render the notebook using Clay (assuming you have the Quarto CLI [installed](https://quarto.org/docs/get-started/)):
```clj
(require '[scicloj.clay.v2.api :as clay])
(clay/make! {:format [:quarto :html]
             :source-path "notebooks/index.clj"})
```

### CODE REVIEW

**Comment by daslu** [comment]
The Tablecloth docs are now rendering mostly correctly through the proposed namespace-as-a-notebook approach -- served temporarily here: https://daslu.github.io/tablecloth/
Since we are rendering it through Quarto, the R code blocks are still evaluated as they used to be in the Rmarkdown docs.
E.g. here: https://daslu.github.io/tablecloth/#data.table

We cannot merge the PR yet since the tests are failing. I'll update soon.

**Comment by daslu** [comment]
The tests pass now, so the PR can be merged from my perspective.

File: .dir-locals.el [hunk 1]
```diff
@@ -0,0 +1,5 @@
+((nil
+  .
+  ((cider-clojure-cli-aliases
+    .
+    "dev"))))
```
Comment (ezmiller) [inline .dir-locals.el:5]: @daslu  What is this doing?

File: README.md [hunk 2]
```diff
@@ -158,6 +158,25 @@ run it before making documentation
...
+- [notebooks/index.clj](notebooks/index.clj) - the tutorial as a Kindly notebook (developed with Clay)
+  - [dev/conversion.clj](dev/conversion.clj) - the script used to generate the notebook from the original `Rmarkdown` tutorial (up to a few additional manual edits)
+- [docs/index.html](docs/index.html) - the tutorial rendered using Clay and [Quarto](https://quarto.org/)
+
+### Actions
+- to render the notebook using Clay (assuming you have the Quarto CLI [installed](https://quarto.org/docs/get-started/)):
+```clj
+(require '[scicloj.clay.v2.api :as clay])
+(clay/make! {:format [:quarto :html]
+             :source-path "notebooks/index.clj"})
```
Comment (ezmiller) [inline README.md:177]: @daslu I think this line needs updating since you renamed to `draft.clj`.

**Review by ezmiller — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (daslu) [reply README.md]: done, thanks

File: .dir-locals.el (hunk 1 above)
Comment (daslu) [reply .dir-locals.el:5]: Making sure the `:dev` alias is used from `deps.edn`, so we have the Clay dependency in our dev environment.

**Comment by daslu** [comment]
Quoting @genmeblog on [the Zulip conversation](https://clojurians.zulipchat.com/#narrow/stream/236259-tech.2Eml.2Edataset.2Edev/topic/tablecloth.20namespace-as-a-notebook.20documentation):

since my time is limited this week I can't move forward. I don't want to stop you so here is my proposal what to do with merging PR:

* I want to keep current docs reachable somewhere (all artifacts) - so let's move it to the "old" subfolder and update all links (readme + github itself)
* then merge Daniel's branch to enable new docs, also link to it somewhere in Readme.
* since the merge we do not update RMarkdown anymore, changes in docs will be done only in Daniel's version
* 
You have all rights to make the merge - feel free to do this

**Pull request merged**